src/context.rs

```diff
@@ -1729,6 +1762,16 @@ impl Context {
     fn optional_session_3(&self) -> ESYS_TR {
         Session::handle_from_option(self.sessions.2).into()
     }
+
+    /// Function that returns the required
+    /// session handle if it is available else
+    /// returns an error.
+    fn required_session_1(&self) -> Result<ESYS_TR> {
+        self.sessions.0.map(|v| v.handle().into()).ok_or_else(|| {
+            error!("Missing session handle for authorization (authSession1 = None)");
+            Error::local_error(ErrorKind::MissingAuthSession)
+        })
+    }
```

Comment:
I wonder if we need to document the fact that we rely on the first session in that tuple whenever a session is _required_ - what do you think?

Reply:
Yes, I think that is needed. That is something we should do for set_sessions too. And refer to the ESAPI specification.